tplinkrouterc6u/client.py

```diff
@@ -1428,11 +1429,509 @@ def _prepare_data(self, data: str, is_login: bool) -> tuple[str, str]:
+            headers['TokenID'] = self._token
+
+        # encrypt request data if needed (for the /cgi_gdpr endpoint)
+        if encrypt:
+            sign, data = self._prepare_data(data_str, False)
+            data = 'sign={}\r\ndata={}\r\n'.format(sign, data)
+        else:
+            data = data_str
+
+        try:
+            retry = 0
+            while retry < self.REQUEST_RETRIES:
+                
+                # send the request
+                if method == 'POST':
+                    r = self.req.post(url, data=data, headers=headers, timeout=self.timeout, verify=self._verify_ssl)
+                elif method == 'GET':
+                    r = self.req.get(url, data=data, headers=headers, timeout=self.timeout, verify=self._verify_ssl)
+                else:
+                    raise Exception('Unsupported method ' + str(method))
+                
+                # sometimes we get 500 here, not sure why... just retry the request
+                if r.status_code != 500 and '<title>500 Internal Server Error</title>' not in r.text:
+                    break
+
+                sleep(0.05)
+                retry += 1
+        except ConnectionError:
+            return 401, "Erro de conexão: A conexão foi abortada ou não pôde ser estabelecida."
+
+        except Timeout:
+            return 408, "Erro: O tempo de conexão ou resposta expirou."
+
+        except RequestException as e:
+            return 403,  f"Erro ao fazer a requisição: {e}"
+
+        # decrypt the response, if needed
+        if encrypt and (r.status_code == 200) and (r.text != ''):
+            return r.status_code, self._encryption.aes_decrypt(r.text)
+        else:
+            return r.status_code, r.text
+
+    def _parse_ret_val(self, response_text):
+        '''
+        Parses $.ret value from the response text
+
+        Return value:
+            return code (int)
+        '''
+        result = search(r'\$\.ret=(.*);', response_text)
+        assert result is not None
+        assert result.group(1).isnumeric()
+
+        return int(result.group(1))
+
+    def _prepare_data(self, data: str, is_login: bool) -> tuple[str, str]:
+        encrypted_data = self._encryption.aes_encrypt(data)
+        data_len = len(encrypted_data)
+        # get encrypted signature
+        signature = self._encryption.get_signature(int(self._seq) + data_len, is_login, self._hash, self._nn, self._ee)
+
+        # format expected raw request data
+        return signature, encrypted_data
+
+
 class TplinkRouterProvider:
     @staticmethod
     def get_client(host: str, password: str, username: str = 'admin', logger: Logger = None,
                    verify_ssl: bool = True, timeout: int = 30) -> AbstractRouter:
-        for client in [TplinkC5400XRouter, TPLinkMRClient, TplinkC6V4Router, TPLinkDecoClient, TplinkRouter]:
+        for client in [TPLinkEXClient, TplinkC5400XRouter, TPLinkMRClient, TplinkC6V4Router, TPLinkDecoClient, TplinkRouter]:
```

Comment:
Could you change it to 
```
for client in [TplinkC5400XRouter, TPLinkMRClient, TPLinkEXClient, TplinkC6V4Router, TPLinkDecoClient, TplinkRouter]:
```
and test for your router TplinkRouterProvider.get_client(...) as there is a possibility that requests for TPLinkEXClient may take the main role before TPLinkMRClient. As we dont have now MR router - so it is better to test it in reverse order

Reply:
Would be great if you could add tests test_client_ex.py. Thank you!
In your client method supports calls self._get_url('cgi/getGDPRParm'). In MR client method supports calls self._get_url('cgi/getParm'), So your router can response on self._get_url('cgi/getGDPRParm') and self._get_url('cgi/getParm') with same response?

Adding routers model to the method supports - greatly reduces the possibilities. As I was very surprised how many different types routers have same API

Reply:
1 - I added the tests (test_client_ex.py)
2 - Yes, the router I tested on responds to "cgi/getGDPRParm" and "cgi/getParm" in the same way.
3 - I believe this new model will support several other routers

Looking at the oid_str.js file, I noticed that there are possibly several other routers that may be supported.

Note the lines that start with INCLUDE_SPEC_

I will test it on a friend's EX220 soon.

I am attaching the file.
[oid_str.txt](https://github.com/user-attachments/files/17710044/oid_str.txt)

Reply:
1 - Thank you for the tests
2 - Thats bad. Because this may break the MR support. I will try to find MR series router - to check response to "cgi/getGDPRParm". Hope it doesnt have

Reply:
I believe this won't be a problem because I adjusted the supports method so that it only responds positively for the EX511 and EX220 models. 

Before, the verification was only by requesting getGDPRParm, but now, in addition to that, the router needs to be on the list of supported models.

However, it would be great to test on a MR, because if there is no conflict, we could simplify the code so that it could support models that we haven't tested, but that use the same API.

If you need to change anything else, let me know.